api_server.py

```
@@ -308,6 +308,49 @@ async def ingest_document(
+            title=f"Ingested {stats.get('successful', 0)} of {stats.get('total_files', 0)} papers",
+            processing_time=processing_time
+        )
+    
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Error ingesting all documents: {e}", exc_info=True)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Error ingesting all documents: {str(e)}"
+        )
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)